src/jrd/replication/Config.cpp

```diff
@@ -216,7 +220,69 @@ Config* Config::get(const PathName& lookupName)
 
 				if (key == "sync_replica")
 				{
-					config->syncReplicas.add(value);
+					SyncReplica syncReplica(config->getPool());
+					if (el.sub)
+					{
+						syncReplica.database = value;
+						for (const auto& sub_el : el.sub->getParameters())
+						{
+							string sub_key(sub_el.name.c_str());
+							string sub_value(sub_el.value);
+
+							if (sub_value.isEmpty())
+								continue;
+
+							auto pos = sub_key.rfind('_');
+							if (pos != string::npos)
+							{
+								const string key_source = sub_key.substr(pos + 1);
+								sub_key = sub_key.substr(0, pos);
+
+								if (key_source.equals(KEY_ENV))
+								{
+									fb_utils::readenv(sub_value.c_str(), sub_value);
+									if (sub_value.isEmpty())
+										configError("missing environment variable", value, sub_value);
+								}
+								else if (key_source.equals(KEY_FILE))
+								{
+									const PathName sub_filename =
+										fb_utils::getPrefix(IConfigManager::DIR_CONF, sub_value.c_str());
```

Comment:
What if the specified value is an absolute pathname? AFAIU, it will be prefixed anyway and become wrong.

Reply:
Fixed, since the configs are stored in the root directory, I don't to add a file check and a separate prefix addition.

Reply:
I'd suggest using `PathUtils::isRelative()` and call `getPrefix()` for relative paths and leave absolute paths "as is".
Also I believe `PathUtils::fixupSeparators()` would be worth adding before opening the file.